avocado/utils/dmesg.py

```diff
@@ -115,3 +122,19 @@ def collect_errors_by_level(output_file=None, level_check=5):
             err += " Please check  dmesg log in debug log."
             LOGGER.debug(dmsg_log)
         raise DmesgError("Test is failed {}".format(err))
+
+
+def skip_dmesg_msg(dmesg_messages, skip_dmesg):
+    """Remove dmesg messages which want to skip.
+
+    :dmesg_messages: dmesg messages from which want to filter
+    :type dmesg_messages: string
+    :skip_dmesg: list of dmesg messages for remove
+    :type skip_dmesg: list
+    """
+    for dmesg in skip_dmesg:
+        for line in dmesg_messages.splitlines():
+            if dmesg in line:
```

Comment:
I could not run this code, since `dmesg_messages` (`process.stdout`) is a byte buffer object, and each `line` will be a byte as well. So unless I"m missing something, this needs a decode/encode properly.

```python
TypeError: a bytes-like object is required, not 'str'
```

Could you paste a basic use case here or on pastebin? A simple call on how to call this method?

Reply:
here no need to decode/encode. i can able to run,not getting any error.this is just a sample how called form a test.
```
error = ['uevent: failed to send synthetic uevent', 'failed to send uevent']
dmesg.collect_errors_by_level(skip_error=error)
```

Reply:
I'm talking about the `skip_dmesg_msg()` method itself. If it is a public method how users will call this? Can you give an example?

Reply:
@smruti77 I missed that you are using `stdout_text` instead of `stdout`. This explains my error.

But I would recommend using `stdout` instead of text to avoid character issues related to `encode`.

Also in your example, shows why we should call this variable `skip_errors` instead of `skip_error`, since this is a list of errors.

Reply:
```stdout_text``` we are using to avoid  ```encode```, ```decode``` error .  and  in ```skip_dmesg_msg()``` if someone want to pass ```stdout```,can pass with ```stdout.decode("utf-8")```. what i feel no need to handle ```encode/decode``` inside  the ```skip_dmesg_msg()```

Reply:
@smruti77 I just double-checked and `avocado.utils.process.run()` is calling `subprocess.run()` with the proper encoding. So, I'm ok with not doing the encoding here.

However, I still have a few suggestions:

1. s/skip_dmesg/skip_messages/g: according to your example, we are going to skip any string here, not only entire dmesg lines.

2. Your regex is not removing properly the empty lines, lefting a few blank lines. Take a very generic example, for instance with `skip_messages = ['usb', 'USB']` and you will see that multiple blank lines will be returned. This is because you are doing the replace on `dmesg_stdout`, and from the `re` PoV there are no multiple lines in this buffer. The EOL (`$`) will be only present the end of the buffer.

3. You don't need the last variable `dmesg_out`, since you can return directly:
```python
return re.sub()...
```
So, in order to filter lines, without leaving blank lines, I would go with something like this (feel free to adap):

```python
  def skip_dmesg_messages(dmesg_stdout, skip_messages):
      """Remove some messages from a dmesg buffer.
  
      This method will remove some lines in a dmesg buffer if some strings are
      present. Returning the same buffer, but with less lines (in case of match).
  
      :dmesg_stdout: dmesg messages from which filter should be applied. This
                     must be a decoded output buffer with new lines.
      :type dmesg_stdout: string
      :skip_messages: list of strings to be removed
      :type skip_messages: list
      """
      def filter_strings(line):
          return not any([string in line for string in skip_messages])
  
      return '\n'.join(filter(None,
                              filter(filter_strings,
                                     dmesg_stdout.splitlines())))

```

Thank you for your patience here.

Reply:
@beraldoleal thak you for suggestion. for space removing i have changed the code,if that is not fine, i will go with this approach.